website/server/libs/auth/index.js

```diff
@@ -128,7 +128,11 @@ async function registerLocal (req, res, { isV3 = false }) {
   }, { 'auth.local': 1 }).exec();
 
   if (user) {
-    if (email === user.auth.local.email) throw new NotAuthorized(res.t('emailTaken'));
+    if (existingUser) {
+      if (email === user.auth.local.email && existingUser._id !== user._id) throw new NotAuthorized(res.t('emailTaken'));
+    } else {
```

Comment:
I think this is what's causing the test failures--
`user` comes from an `$or` search, either we find the email address or we find the lowercase username;
but if we find it on the lowercase username, we end up here, reporting that the email address is already taken even if the email address passed the uniqueness check.